File: examples/node_health.exs
# OpenClaw Node Health Check (Elixir)
defmodule NodeHealth do
  def check(url) do
    :inets.start()
    :ssl.start()
    request = {String.to_charlist(url <> "/health"), []}

    case :httpc.request(:get, request, [timeout: 5000], []) do
      {:ok, {{_version, status, _reason}, _headers, _body}} ->
        "OK   #{url}: #{status}"

      {:error, reason} ->
        "FAIL #{url}: #{inspect(reason)}"
    end
  end
end

nodes =
  case System.argv() do
    [] -> ["http://localhost:8080", "http://localhost:8081"]
    args -> args
  end

Enum.each(nodes, fn node -> IO.puts(NodeHealth.check(node)) end)

File: examples/node_health.erl
%% OpenClaw Node Health Check (Erlang) -- via httpc/inets
-module(node_health).
-export([main/1, check/1]).

check(Url) ->
    inets:start(),
    ssl:start(),
    Request = {Url ++ "/health", []},
    case httpc:request(get, Request, [{timeout, 5000}], []) of
        {ok, {{_Version, Status, _Reason}, _Headers, _Body}} ->
            io_lib:format("OK   ~s: ~p", [Url, Status]);
        {error, Reason} ->
            io_lib:format("FAIL ~s: ~p", [Url, Reason])
    end.

main(Args) ->
    Nodes =
        case Args of
            [] -> ["http://localhost:8080", "http://localhost:8081"];
            _ -> Args
        end,
    lists:foreach(fun(Node) -> io:format("~s~n", [check(Node)]) end, Nodes).
